internal/primitives/core/crypto/crypto.go

```diff
@@ -0,0 +1,246 @@
+package crypto
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hashing"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// The root phrase for our publicly known keys.
+const DevPhrase = "bottom drive obey lake curtain smoke basket hold race lonely fit walk"
+
+// A since derivation junction description. It is the single parameter used when creating
+// a new secret key from an existing secret key and, in the case of `SoftRaw` and `SoftIndex`
+// a new public key from an existing public key.
+type DeriveJunction struct {
```

Comment:
forgive me if this is a naive question, but what is a function in this sense?

Reply:
you mean junction?

Reply:
oops, yeah sorry I meant junction, not function

Reply:
Yea, it's just the steps on how this key was derived.